File: php/09proc/proc60.php
<?php
function Leng($xA, $yA, $xB, $yB){
		return number_format(sqrt(pow($xA - $xB, 2) + pow($yA - $yB, 2)), 2);
}
function Area($xA, $yA, $xB, $yB, $xC, $yC){
        $a = Leng($xA, $yA, $xB, $yB);
        $b = Leng($xB, $yB, $xC, $yC);
        $c = Leng($xA, $yA, $xC, $yC);
        $P = $a + $b + $c;
        $p = $P / 2;
        $S = sqrt($p * ($p - $a) * ($p - $b) * ($p - $c));
		$S = number_format($S, 2);
        return $S;
}
function Dist($xP, $yP, $xA, $yA, $xB, $yB){
		return number_format(2 * Area($xP, $yP, $xA, $yA, $xB, $yB) / Leng($xA, $yA, $xB, $yB), 2);
}
function Heights($xA, $yA, $xB, $yB, $xC, $yC, &$HA, &$HB, &$HC){
		$HA = Dist($xA, $yA, $xC, $yC, $xB, $yB);
        $HB = Dist($xB, $yB, $xC, $yC, $xA, $yA);
        $HC = Dist($xC, $yC, $xA, $yA, $xB, $yB);
}
$xA = mt_rand(1, 100);
$yA = mt_rand(1, 100);
$xB = mt_rand(1, 100);
$yB = mt_rand(1, 100);
$xC = mt_rand(1, 100);
$yC = mt_rand(1, 100);
$xD = mt_rand(1, 100);
$yD = mt_rand(1, 100);
echo "<br>xA = ".$xA;
echo "<br>yA = ".$yA;
echo "<br>xB = ".$xB;
echo "<br>yB = ".$yB;
echo "<br>xC = ".$xC;
echo "<br>yC = ".$yC;
echo "<br>xD = ".$xD;
echo "<br>yD = ".$yD;
Heights($xA, $yA, $xB, $yB, $xC, $yC, $HA, $HB, $HC);
echo "<br><br>Секунҷаи ABC:";
echo "<br>HA = ".$HA;
echo "<br>HB = ".$HB;
echo "<br>HC = ".$HC;
Heights($xA, $yA, $xB, $yB, $xD, $yD, $HA, $HB, $HC);
echo "<br><br>Секунҷаи ABD:";
echo "<br>HA = ".$HA;
echo "<br>HB = ".$HB;
echo "<br>HD = ".$HC;
Heights($xB, $yB, $xC, $yC, $xD, $yD, $HA, $HB, $HC);
echo "<br><br>Секунҷаи BCD:";
echo "<br>HB = ".$HA;
echo "<br>HC = ".$HB;
echo "<br>HD = ".$HC;
?>

File: cpp/09proc/Proc60.cpp
#include <iostream>
#include <cmath>
using namespace std;

struct Point
{
	double x, y;
};

void Altitudes(double xA, double yA, double xB, double yB, double xC, double yC,
			   double& hA, double& hB, double& hC);
double Dist(double xP, double yP, double xA, double yA, double xB, double yB);
double Area(double xA, double yA, double xB, double yB, double xC, double yC);
double Perim(double xA, double yA, double xB, double yB, double xC, double yC,
			 double& a, double& b, double& c);
double Leng(double xA, double yA, double xB, double yB);

int main()
{
	//Task("Proc60");
	Point A, B, C, D;

	cin >> A.x >> A.y;
	cin >> B.x >> B.y;
	cin >> C.x >> C.y;
	cin >> D.x >> D.y;

	double hA, hB, hC;
	Altitudes(A.x, A.y, B.x, B.y, C.x, C.y, hA, hB, hC);
	cout << hA << hB << hC;

	Altitudes(A.x, A.y, B.x, B.y, D.x, D.y, hA, hB, hC);
	cout << hA << hB << hC;

	Altitudes(A.x, A.y, C.x, C.y, D.x, D.y, hA, hB, hC);
	cout << hA << hB << hC;
	
	return 0;
}

void Altitudes(double xA, double yA, double xB, double yB, double xC, double yC,
			   double& hA, double& hB, double& hC)
{
	hA = Dist(xA, yA, xB, yB, xC, yC);
	hB = Dist(xB, yB, xA, yA, xC, yC);
	hC = Dist(xC, yC, xA, yA, xB, yB);
}

double Dist(double xP, double yP, double xA, double yA, double xB, double yB)
{
	return 2 * Area(xP, yP, xA, yA, xB, yB) / Leng(xA, yA, xB, yB);
}

double Area(double xA, double yA, double xB, double yB, double xC, double yC)
{
	double a, b, c;
	double p = Perim(xA, yA, xB, yB, xC, yC, a, b, c) / 2;

	return sqrt( p*(p-a)*(p-b)*(p-c) );
}

double Perim(double xA, double yA, double xB, double yB, double xC, double yC,
			 double& a, double& b, double& c)
{
	a = Leng(xA, yA, xB, yB);
	b = Leng(xA, yA, xC, yC);
	c = Leng(xB, yB, xC, yC);

	return a+b+c;
}

double Leng(double xA, double yA, double xB, double yB)
{
	return sqrt( pow(xB - xA, 2) + pow(yB - yA, 2) );
}